components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -282,6 +306,61 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_date_diff() {
+        let cases = vec![
+            (
+                "0000-01-01 00:00:00.000000",
+                "0000-01-01 00:00:00.000000",
+                0,
+            ),
+            (
+                "2018-02-01 00:00:00.000000",
+                "2018-02-01 00:00:00.000000",
+                0,
+            ),
+            (
+                "2018-02-02 00:00:00.000000",
+                "2018-02-01 00:00:00.000000",
+                1,
+            ),
+            (
+                "2018-02-01 00:00:00.000000",
+                "2018-02-02 00:00:00.000000",
+                -1,
+            ),
+            (
+                "2018-02-02 00:00:00.000000",
+                "2018-02-01 23:59:59.999999",
+                1,
+            ),
+            (
+                "2018-02-01 23:59:59.999999",
+                "2018-02-02 00:00:00.000000",
+                -1,
+            ),
+        ];
+
+        let mut ctx = EvalContext::default();
```

Comment:
Please add some cases to cover invalid zero time and allow invalid time (e.g non-strict mode). thanks